smoke-test.go

```diff
@@ -0,0 +1,279 @@
+	alarmDefActions := []string{notificationID}
+	requestBody := models.AlarmDefinitionRequestBody{Name: &name, Expression: &expression, AlarmActions: &alarmDefActions, UndeterminedActions: &alarmDefActions, OkActions: &alarmDefActions}
+	alarmDefinitionResponse, err := monascaclient.CreateAlarmDefinition(&requestBody)
+	if err != nil {
+		fmt.Printf("FAILED - Error creating alarm definition %s\n", err.Error())
+		return ""
+	}
+	fmt.Println("SUCCESS")
+	testsSucceeded++
+	return alarmDefinitionResponse.ID
+}
+
+func testCreateMetric(value float64) {
+	now := time.Now().Round(time.Millisecond).UnixNano() / (int64(time.Millisecond) / int64(time.Nanosecond))
+	name := "smoke_test_metric"
+	err := monascaclient.CreateMetric(nil, &models.MetricRequestBody{Name: &name, Value: &value, Timestamp: &now})
+	if err != nil {
+		fmt.Printf("FAILED - Error creating metric %s\n", err.Error())
+		return
+	}
+	fmt.Println("SUCCESS")
+	testsSucceeded++
+}
+
+func testWebhookTrigger() {
+	// Wait 5 minutes for webhook to trigger
+	for i := 0; i < 10; i++ {
+		if webhookTriggered {
+			fmt.Println("SUCCESS")
+			testsSucceeded++
+			return
+		}
+		time.Sleep(3 * time.Second)
+	}
+	fmt.Println("FAILED - Did not recieve webhook")
+}
+
+func cleanup(alarmDefinitionID, notificationID string) {
+	cleanupStatus := true
+	if alarmDefinitionID != "" {
+		err := monascaclient.DeleteAlarmDefinition(alarmDefinitionID)
+		if err != nil {
+			fmt.Printf("FAILED - Error alarm definition - %s\n", err.Error())
+			cleanupStatus = false
+		}
+	}
+	if notificationID != "" {
+		err := monascaclient.DeleteNotificationMethod(notificationID)
+		if err != nil {
+			fmt.Printf("FAILED - Error deleting alarm definition - %s\n", err.Error())
+			cleanupStatus = false
+		}
+	}
+	if cleanupStatus {
+		testsSucceeded++
+		fmt.Println("SUCCESS")
+	}
+}
+
+func cleanupPreviousRun() {
+	notificationName := "smoke_test_notification"
+	// Get notification method
+	notificationMethods, err := monascaclient.GetNotificationMethods(&models.NotificationQuery{})
+	if err != nil {
+		fmt.Printf("Error getting notification methods to delete potential left over notification method "+
+			"from previous runs - %s\n", err.Error())
+	} else {
+		notificationID := ""
+		for _, notificationMethod := range notificationMethods.Elements {
+			if notificationMethod.Name == notificationName {
+				notificationID = notificationMethod.ID
+			}
+		}
+		if notificationID != "" {
+			err := monascaclient.DeleteNotificationMethod(notificationID)
+			if err != nil {
+				fmt.Printf("Error deleting notification method - %s\n", err.Error())
+			}
+		}
+	}
+	alarmDefinitionName := "smoke_test_alarm"
+	alarmDefinitions, err := monascaclient.GetAlarmDefinitions(&models.AlarmDefinitionQuery{Name: &alarmDefinitionName})
+	if err != nil {
+		fmt.Printf("Error getting alarm definitions to delete potential left over alarm definition "+
+			"from previous runs - %s\n", err.Error())
+	} else {
+		alarmDefinitionID := ""
```

Comment:
There should be 0 or 1 elements. Nothing to do if 0, if 1 delete it. Would be simpler